src/main/java/novamachina/exnihilosequentia/common/tileentity/barrel/AbstractBarrelTile.java

```diff
@@ -76,14 +107,18 @@ public void tick() {
         }
 
         if (mode.isEmptyMode() || mode.getModeName().equals(ExNihiloConstants.BarrelModes.FLUID)) {
-            BlockPos abovePos = worldPosition.offset(0, 1, 0);
-            if (getLevel().isRainingAt(abovePos)) {
-                FluidStack stack = new FluidStack(Fluids.WATER, Config.getRainFillAmount());
+            final BlockPos abovePos = worldPosition.offset(0, 1, 0);
+            if (level.isRainingAt(abovePos) && tank.getSpace() >= Config.getRainFillAmount()) {
```

Comment:
Forgot to mention I also check the space to prevent the creation of a new FluidStack.